### PR TITLE
Bugfix/mobile friendly header + auth directives

### PR DESCRIPTION
**Pull request by 99999989**
- create auth directives: alvHasAnyAuthority and alvIsAuthenticated
- use it to hide the menu + menu toggle button
- prettify the header for mobile

### CODE REVIEW

File: alv-portal-ui/src/app/core/auth/has-any-authority/has-any-authority.directive.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import { Directive, Input, TemplateRef, ViewContainerRef } from '@angular/core';
+import { AuthenticationService } from '../../authentication/authentication.service';
+
+@Directive({
+  selector: '[alvHasAnyAuthority]'
+})
+export class HasAnyAuthorityDirective {
+
+  private _hasAnyAuthority: string | Array<string>;
```
Comment (pavax) [inline alv-portal-ui/src/app/core/auth/has-any-authority/has-any-authority.directive.ts]: remove the _

File: alv-portal-ui/src/app/core/auth/has-any-authority/has-any-authority.directive.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  constructor(private authenticationService: AuthenticationService,
+              private templateRef: TemplateRef<any>,
+              private viewContainerRef: ViewContainerRef) {
+  }
+
+  @Input()
+  set alvHasAnyAuthority(value: string | Array<string>) {
+    this._hasAnyAuthority = typeof value === 'string' ? [<string> value] : <string[]> value;
+    this.authenticationService.getCurrentUser()
+        .subscribe(user => {
```
Comment (pavax) [inline alv-portal-ui/src/app/core/auth/has-any-authority/has-any-authority.directive.ts]: how do we make sure that this directive unsubscribes itself after it has been destroyed?

Comment (99999989) [reply alv-portal-ui/src/app/core/auth/has-any-authority/has-any-authority.directive.ts]: We use the takeUntil approach (I've updated the code). 
But we could also create our own decorators like here: https://netbasal.com/automagically-unsubscribe-in-angular-4487e9853a88

File: alv-portal-ui/src/app/core/authentication/user.model.ts [hunk 3]
```diff
@@ -8,6 +8,15 @@ export interface User {
   langKey: string;
   authorities: Array<string>;
   registrationStatus: RegistrationStatus;
+
+  containsAuthority(authorities: string | Array<string>): boolean {
+    for (let i = 0; i < authorities.length; i++) {
+      if (this.authorities.includes(authorities[i])) {
```
Comment (ganqqwerty) [inline alv-portal-ui/src/app/core/authentication/user.model.ts]: can you please use this function here as well? alv-portal-ui/src/app/core/landing-navigation.service.ts

**Pull request merged**